ext/IntervalArithmeticForwardDiffExt.jl

```diff
@@ -3,7 +3,19 @@ module IntervalArithmeticForwardDiffExt
 using IntervalArithmetic, ForwardDiff
 using ForwardDiff: Dual, ≺, value, partials
 
-#
+# Needed to avoid method ambiguities:
+Base.promote_rule(::Type{Dual{T, V, N}}, ::Type{Interval{S}}) where {T, V, N, S<:Union{AbstractFloat, Rational}} =
+    Interval{IntervalArithmetic.promote_numtype(V, S)}
```

Comment:
Not sure if this should return `Interval` or `Dual{T, Interval{IntervalArithmetic.promote_numtype(V, S), N}`...?

@OlivierHnt Any ideas?

Same for `ExactReal`...

Reply:
Unfortunately I am note familiar with ForwardDiff.jl... My first impulse would be `Dual{T, Interval{IntervalArithmetic.promote_numtype(V, S), N}`.
Maybe @Kolaru knows about this more? (though I believe he is very busy at the moment with the PhD defence)

Reply:
Locally, tests pass (though I do not get the messages on the ambiguities) using `Dual{T, Interval{IntervalArithmetic.promote_numtype(V, S), N}` too. I thought about this after pushing the commit, and checking the methods defined for Dual's...